lib/spack/spack/test/conftest.py

```diff
@@ -63,6 +64,59 @@ def no_stdin_duplication(monkeypatch):
                         lambda x: StringIO())
 
 
+@pytest.fixture()
+def spec_from_dict():
+    """Returns a factory that builds a Spec from a dictionary. This allows
+    the construction of a DAG with literals in tests.
+
+    The dictionary must have a single top level key, representing the root,
+    and as many secondary level keys as needed in the spec. The definition
+    is recursive.
+
+    Examples:
+        A simple spec ``foo`` with no dependencies:
+
+            {'foo': None}
+
+        A spec ``foo`` with a ``(build, link)`` dependency ``bar``:
+
+            {'foo':
+                {'bar:build,link': None}}
+
+    """
+    def _impl(spec_dict):
+
+        # The invariant is that the top level dictionary must have
+        # only one key
+        assert len(spec_dict) == 1
```

Comment:
I actually think it'd be useful if it didn't have to have one key.  A dict of multiple specs could be useful, but this can wait since it's not necessary.